packages/metal-clay-button/src/MetalClayButton.soy

```diff
@@ -17,27 +17,56 @@
 
 	{if $href}
 		<a href="{$href}" {call .attributes data="all" /}>
-			{call .icon data="all" /}
-			{$label or ''}
+			{call .content}
+				{param icon: $icon /}
+				{param label: $label /}
+			{/call}
 		</a>
 	{else}
 		<button {call .attributes data="all" /}>
-			{call .icon data="all" /}
-			{$label or ''}
+			{call .content}
+				{param icon: $icon /}
+				{param label: $label /}
+			{/call}
 		</button>
 	{/if}
 {/template}
 
-{template .icon}
+/**
+ * Renders element content
+ */
+{template .content}
 	{@param? icon: ?}
+	{@param? label: html}
 
-	{if $icon}
-		{call MetalClayIcon.render}
-			{param elementClasses: $icon.elementClasses /}
-			{param spritemap: $icon.spritemap /}
-			{param symbol: $icon.symbol /}
+	{let $iconAlignment: $icon and $icon.alignment ? $icon.alignment : 'left' /}
```

Comment:
I'm not sure if it will work here once we have a logical operation for checking `$icon` and `$icon.alignment`.